sdk/tracing/azure-core-tracing-opencensus/src/samples/java/com/azure/core/tracing/opencensus/ListOperationsAsync.java

```diff
@@ -0,0 +1,89 @@
+     * using a zipkin exporter.
+     *
+     * @param args Unused. Arguments to the program.
+     * @throws IllegalArgumentException when invalid key vault endpoint is passed.
+     */
+    public static void main(String[] args) throws IllegalArgumentException {
+        ZipkinTraceExporter.createAndRegister("http://localhost:9411/api/v2/spans", "tracing-to-zipkin-service");
+
+        TraceConfig traceConfig = Tracing.getTraceConfig();
+        TraceParams activeTraceParams = traceConfig.getActiveTraceParams();
+        traceConfig.updateActiveTraceParams(activeTraceParams.toBuilder().setSampler(Samplers.alwaysSample()).build());
+
+        Tracer tracer = Tracing.getTracer();
+
+        // Instantiate a client that will be used to call the service. Notice that the client is using default Azure
+        // credentials. To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
+        // 'AZURE_CLIENT_KEY' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+        SecretAsyncClient client = new SecretClientBuilder()
+            .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+            .credential(new DefaultAzureCredentialBuilder().build())
+            .buildAsyncClient();
+
+        Scope scope = tracer.spanBuilder("user-parent-span").startScopedSpan();
+        try {
+            Context traceContext = Context.of(OPENCENSUS_SPAN_KEY, tracer.getCurrentSpan());
+            // Let's create secrets holding storage and bank accounts credentials. if the secret
+            // already exists in the key vault, then a new version of the secret is created.
+            client.setSecret(new Secret("StorageAccountPassword", "password"))
+                .then(client.setSecret(new Secret("BankAccountPassword", "password")))
+                .subscriberContext(traceContext)
+                .block();
+
+            // You need to check if any of the secrets are sharing same values. Let's list the secrets and print their values.
+            // List operations don't return the secrets with value information. So, for each returned secret we call getSecret to get the secret with its value information.
+            client.listSecrets()
+                .subscriberContext(traceContext)
+                .subscribe(secretBase -> client.getSecret(secretBase)
+                    .subscriberContext(traceContext)
+                    .subscribe(secret -> System.out.printf("Received secret with name %s and value %s%n",
+                        secret.getName(), secret.getValue())));
+
+            // The bank account password got updated, so you want to update the secret in key vault to ensure it reflects the new password.
+            // Calling setSecret on an existing secret creates a new version of the secret in the key vault with the new value.
+            client.setSecret("BankAccountPassword", "new password")
```

Comment:
Using .block() makes this sample a synchronous one and does not showcase how to use it asynchronously.

Reply:
removed.